pyne/r2s.py

```diff
@@ -230,31 +232,107 @@ def total_photon_source_intensity(m, tag_name, sub_voxel=False):
-        The energy boundaries of a photon source.
+        The mesh instance to be tagged.
+    e_bounds: iterable of float
+        The energy boundaries of a photon source, eV.
     tag_name : str, optional
-       The name of the tag on the mesh with the energy boundaries
-       information.
+       The name of the energy boundaries tag.
 
     Returns
     -------
     m : PyNE Mesh
        The mesh with energy boundaries tag.
     """
+
     # do not provide value when init a sparse tag
-    m.tag(name=tag_name, doc='energy boundaries of the photon source',
+    m.tag(name=tag_name, doc=tag_name + ' of the photon source',
           tagtype=NativeMeshTag, size=len(e_bounds), dtype=float,
           storage_type='sparse')
     m.get_tag(tag_name)[m] = e_bounds
     return m
 
 
+def tag_source_intensity(m, source_intensity, tag_name='source_intensity'):
+    """This function tags the source intensity of photon source to the PyMOAB
+    mesh instance as a sparse tag for the purpose of photon source sampling.
+
+    Parameters
+    ----------
+    m : PyNE Mesh
+        The object containing the mesh instance to be tagged.
+    source_intensity : float
+        The source intensity of a photon source, p/s.
+    tag_name : str, optional
+        The name of the energy boundaries tag.
+
+    Returns
+    -------
+    m : PyNE Mesh
+        The mesh with energy boundaries tag.
+    """
+
+    # do not provide value when initializing a sparse tag
+    m.tag(name=tag_name, doc=tag_name + ' of the photon source',
+          tagtype=NativeMeshTag, size=1, dtype=float,
+          storage_type='sparse')
+    m.get_tag(tag_name)[m] = source_intensity
+    return m
+
+
+def tag_decay_time(m, decay_time, tag_name='decay_time'):
+    """This function tags the decay time of photon source to the PyMOAB
+    mesh instance as a sparse tag for the purpose of photon source sampling.
+
+    Parameters
+    ----------
+    m : PyNE Mesh
+        The object containing the mesh instance to be tagged.
+    decay_time : float
+        The decay time of a photon source, s.
+    tag_name : str, optional
+        The name of the decay time tag.
+
+    Returns
+    -------
+    m : PyNE Mesh
+        The mesh with decay time tag.
+    """
+
+    # do not provide value when init a sparse tag
+    m.tag(name=tag_name, doc=tag_name + ' of the photon source',
+          tagtype=NativeMeshTag, size=1, dtype=float,
+          storage_type='sparse')
+    m.get_tag(tag_name)[m] = decay_time
+    return m
+
+
+def tag_version(m):
+    """This function tags the version of photon source to the PyMOAB
+    mesh instance as a sparse tag for the purpose of photon source sampling.
+
+    Parameters
+    ----------
+    m : PyNE Mesh
+        The object containing the mesh instance to be tagged.
+
+    Returns
+    -------
+    m : PyNE Mesh
+       The mesh with version tag.
+    """
+
+    # do not provide value when init a sparse tag
+    tag_name = 'source_file_version'
```

Comment:
I'd make this global as well, a `_SOURCE_TAG_NAME` variable at file scope.

Reply:
Done in https://github.com/pyne/pyne/pull/1352/commits/87d08b1dea984a04f6ffe97423f32de870a260c9.